osu.Desktop/Windows/WindowsAssociationManager.cs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using Microsoft.Win32;
+using osu.Framework.Localisation;
+using osu.Framework.Logging;
+using osu.Game.Localisation;
+
+namespace osu.Desktop.Windows
+{
+    [SupportedOSPlatform("windows")]
+    public static class WindowsAssociationManager
+    {
+        public const string SOFTWARE_CLASSES = @"Software\Classes";
+
+        /// <summary>
+        /// Sub key for setting the icon.
+        /// https://learn.microsoft.com/en-us/windows/win32/com/defaulticon
+        /// </summary>
+        public const string DEFAULT_ICON = @"DefaultIcon";
+
+        /// <summary>
+        /// Sub key for setting the command line that the shell invokes.
+        /// https://learn.microsoft.com/en-us/windows/win32/com/shell
+        /// </summary>
+        public const string SHELL_OPEN_COMMAND = @"Shell\Open\Command";
+
+        public static readonly string EXE_PATH = Path.ChangeExtension(typeof(WindowsAssociationManager).Assembly.Location, ".exe").Replace('/', '\\');
```

Comment:
What's with the replace at the end? This already appears to be correct on my machine...

```csharp
> typeof(Program).Assembly.Location
  "D:\\Open Source\\osu\\osu.Desktop\\bin\\Debug\\net8.0\\osu!.dll"

> Path.ChangeExtension(typeof(Program).Assembly.Location, ".exe")
  "D:\\Open Source\\osu\\osu.Desktop\\bin\\Debug\\net8.0\\osu!.exe"
```

Reply:
I dunno that's weird. I'd like to think windows should be able to handle either?